projects/packages/jetpack-mu-wpcom/src/features/override-preview-button-url/override-preview-button-url.js

```diff
@@ -0,0 +1,33 @@
+import { use } from '@wordpress/data';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * The gutenberg block editor preview button opens a new window to a simple site's mapped
+ * domain.
+ * Adds logmein query param to editor draft post preview url to add WordPress cookies in
+ * a first party context ( allowing us to avoid third party cookie issues )
+ */
+async function overridePreviewButtonUrl() {
+	use( registry => {
+		return {
+			dispatch: store => {
+				const namespace = store.name ?? store;
+				const actions = { ...registry.dispatch( namespace ) };
+
+				if ( namespace === 'core/editor' && actions.__unstableSaveForPreview ) {
+					const { __unstableSaveForPreview } = actions;
+					actions.__unstableSaveForPreview = async ( ...args ) => {
+						const link = await __unstableSaveForPreview( ...args );
+						return link.startsWith( window.location.origin )
+							? link
+							: addQueryArgs( link, { logmein: 'direct' } );
```

Comment:
![badge](https://img.shields.io/badge/review-ask-blue.svg)

Is my understanding of `logmein` correct?:

- The logic lives in fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Sybtzrva.cuc%3Se%3Q97ooo5qs-og
- It shortcuts some links and avoids cookie issues fbhepr%2Skers%2Sjcpbz%2Sova%2Sgrfgf%2Svfbyngrq%2Sfhvgrf%2SErzbgrYbtva%2SYbtzrvaGrfg.cuc%3Se%3Qqr14000s%2398-og

Also, it is expected that the final preview URL becomes like `https://<customdomain>/?p=7&preview=true` (without `logmein` param), am I right? I confirmed this line was executed.

Reply:
Yes, that's my understanding as well!